django/core/serializers/base.py

```diff
@@ -283,15 +283,23 @@ def m2m_convert(v):
             return model._meta.pk.to_python(v)
 
     try:
-        values = []
-        for pk in field_value:
-            values.append(m2m_convert(pk))
-        return values
-    except Exception as e:
-        if isinstance(e, ObjectDoesNotExist) and handle_forward_references:
-            return DEFER_FIELD
+        pks_iter = iter(field_value)
+    except TypeError as e:
+        raise M2MDeserializationError(e, field_value)
+
+    values = []
+    for pk in pks_iter:
+        try:
+            converted = m2m_convert(pk)
+        except Exception as e:
+            if isinstance(e, ObjectDoesNotExist) and handle_forward_references:
+                return DEFER_FIELD
+            else:
+                raise M2MDeserializationError(e, pk)
         else:
-            raise M2MDeserializationError(e, pk)
+            values.append(converted)
+
+    return values
```

Comment:
Some parts of this change are unrelated with a bugfix :thinking: Can we leave only necessary changes?, i.e.
```diff
diff --git a/django/core/serializers/base.py b/django/core/serializers/base.py
index cf8e7d3b86..fac4440b62 100644
--- a/django/core/serializers/base.py
+++ b/django/core/serializers/base.py
@@ -282,9 +282,14 @@ def deserialize_m2m_values(field, field_value, using, handle_forward_references)
         def m2m_convert(v):
             return model._meta.pk.to_python(v)

+    try:
+        pks_iter = iter(field_value)
+    except TypeError as e:
+        raise M2MDeserializationError(e, field_value)
     try:
         values = []
-        for pk in field_value:
+        for pk in pks_iter:
             values.append(m2m_convert(pk))
         return values
     except Exception as e:
```

Reply:
Sure, I guess I cleaned up a bit more than what was strictly necessary.